stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -739,85 +728,205 @@ end
 
 #Linear solvers
 #Generic solver using naive substitution
-function ldiv!(A::Bidiagonal, b::AbstractVector)
-    require_one_based_indexing(A, b)
+ldiv!(A::Bidiagonal, b::AbstractVecOrMat) = @inline ldiv!(b, A, b)
+function ldiv!(c::AbstractVecOrMat, A::Bidiagonal, b::AbstractVector)
+    require_one_based_indexing(c, A, b)
     N = size(A, 2)
     mb = length(b)
     if N != mb
         throw(DimensionMismatch("second dimension of A, $N, does not match the length of b, $mb"))
     end
+    mc, nc = size(c, 1), size(c, 2)
+    if mc != mb || nc != 1
+        throw(DimensionMismatch("size of result, ($mc, $nc), does not match the length of b, $mb"))
+    end
 
     if N == 0
-        return b
+        return copyto!(c, b)
     end
 
     @inbounds begin
         if A.uplo == 'L' #do forward substitution
-            b[1] = bj1 = A.dv[1]\b[1]
+            c[1] = bj1 = A.dv[1]\b[1]
             for j in 2:N
-                bj  = b[j]
-                bj -= A.ev[j - 1] * bj1
                 dvj = A.dv[j]
                 if iszero(dvj)
                     throw(SingularException(j))
                 end
+                bj  = b[j]
+                bj -= A.ev[j - 1] * bj1
                 bj   = dvj\bj
-                b[j] = bj1 = bj
+                c[j] = bj1 = bj
             end
         else #do backward substitution
-            b[N] = bj1 = A.dv[N]\b[N]
-            for j = (N - 1):-1:1
-                bj  = b[j]
-                bj -= A.ev[j] * bj1
+            c[N] = bj1 = A.dv[N]\b[N]
+            for j in (N - 1):-1:1
                 dvj = A.dv[j]
                 if iszero(dvj)
                     throw(SingularException(j))
                 end
+                bj  = b[j]
+                bj -= A.ev[j] * bj1
                 bj   = dvj\bj
-                b[j] = bj1 = bj
+                c[j] = bj1 = bj
             end
         end
     end
-    return b
+    return c
 end
-function ldiv!(A::Bidiagonal, B::AbstractMatrix)
-    require_one_based_indexing(A, B)
+function ldiv!(C::AbstractMatrix, A::Bidiagonal, B::AbstractMatrix)
     mA, nA = size(A)
-    n = size(B, 1)
+    n, m = size(B)
     if mA != n
         throw(DimensionMismatch("first dimension of A, $mA, does not match the first dimension of B, $n"))
     end
-    for b in eachcol(B)
-        ldiv!(A, b)
+    mC, nC = size(C)
+    if mC != n || nC != m
+        throw(DimensionMismatch("size of result is ($mC, $nC), but should be ($n, $m)"))
     end
-    B
+    for (c, b) in zip(eachcol(C), eachcol(B))
+        ldiv!(c, A, b)
+    end
+    C
 end
-ldiv!(A::Transpose{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = ldiv!(copy(A), b)
-ldiv!(A::Adjoint{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = ldiv!(copy(A), b)
+ldiv!(A::Transpose{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = @inline ldiv!(b, copy(A), b)
+ldiv!(A::Adjoint{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = @inline ldiv!(b, copy(A), b)
```

Comment:
```suggestion
ldiv!(A::Transpose{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = @inline ldiv!(b, A, b)
ldiv!(A::Adjoint{<:Any,<:Bidiagonal}, b::AbstractVecOrMat) = @inline ldiv!(b, A, b)
```
The `copy(A)` can occur in the 3-arg method if needed.